src/Tasks/Microsoft.Common.CurrentVersion.targets

```
@@ -1824,14 +1824,19 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         much information as possible will be passed to the compilers.
         -->
 
+    <PropertyGroup>
+      <MSBuildGlobalPropertiesToRemoveFromProjectReferences>%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)</MSBuildGlobalPropertiesToRemoveFromProjectReferences>
+      <MSBuildGlobalPropertiesToRemoveFromProjectReferences Condition="'$(DoNotPassOutputPathToReferencedProjects)'=='true'">%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);OutputPath</MSBuildGlobalPropertiesToRemoveFromProjectReferences>
+    </PropertyGroup>
+
     <MSBuild
         Projects="@(_MSBuildProjectReferenceExistent)"
         Targets="GetTargetPath"
         BuildInParallel="$(BuildInParallel)"
         Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform); %(_MSBuildProjectReferenceExistent.SetTargetFramework)"
         Condition="'%(_MSBuildProjectReferenceExistent.BuildReference)' == 'true' and '@(ProjectReferenceWithConfiguration)' != '' and ('$(BuildingInsideVisualStudio)' == 'true' or '$(BuildProjectReferences)' != 'true') and '$(VisualStudioVersion)' != '10.0' and '@(_MSBuildProjectReferenceExistent)' != ''"
         ContinueOnError="!$(BuildingProject)"
-        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)">
+        RemoveProperties="$(MSBuildGlobalPropertiesToRemoveFromProjectReferences)">
```

Does the following work?

```suggestion
        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);$(MSBuildGlobalPropertiesToRemoveFromProjectReferences)">
```

If so, then I think it would fix the issue of collapsing all GlobalPropertiesToRemove metadata from different project references into one list.